external/fv3fit/fv3fit/_shared/xr_prediction.py

```diff
@@ -0,0 +1,72 @@
+import numpy as np
+from typing import Sequence, Iterable, Hashable, Protocol
+import xarray as xr
+
+from fv3fit._shared.halos import append_halos, append_halos_using_mpi
+from fv3fit._shared import (
+    match_prediction_to_input_coords,
+    SAMPLE_DIM_NAME,
+    stack,
+)
+
+
+class Model(Protocol):
```

Comment:
Can you rename this to Predictor, which we used to avoid overloading the "model" name (even if I broke that rule when naming this directory)?

Reply:
I renamed it to`ArrayPredictor` just in case someone looking at the type hint got confused with the actual `Predictor` class, which takes in and outputs datasets.